module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -106,22 +155,32 @@ protected function validateConfig()
     public function authenticate($request)
     {
         // Check if username is set.
-        $shib = $this->getConfig()->Shibboleth;
-        $username = $request->getServer()->get($shib->username);
+        $entityId = $this->getCurrentEntityId($request);
+        $shib = $this->getConfigurationLoader()->getConfiguration($entityId);
+        $username = $this->getAttribute($request, $shib['username']);
+        $proxy = $this->getConfig()->Shibboleth->proxy ?? false;
```

Comment:
I think it would be better to make the $proxy flag a member attribute and set it once in constructor. Now it's read in several places separately.